core/java/com/google/instrumentation/stats/Tag.java

```diff
@@ -38,6 +38,26 @@ public TagValue getValue() {
     return value;
   }
 
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
Guava has a nice utility for equals/hashCode tests com.google.common.testing.EqualsTester;